diesel_derives2/src/as_expression.rs

```diff
@@ -0,0 +1,122 @@
+use quote;
+use syn;
+
+use meta::*;
+use util::*;
+
+pub fn derive(item: syn::DeriveInput) -> Result<quote::Tokens, Diagnostic> {
+    let dummy_mod = format!(
+        "_impl_as_expression_for_{}",
+        item.ident.as_ref().to_lowercase()
+    );
+    let flags =
+        MetaItem::with_name(&item.attrs, "diesel").unwrap_or_else(|| MetaItem::empty("diesel"));
+    let is_sized = !flags.has_flag("not_sized");
+
+    let sql_types = MetaItem::all_with_name(&item.attrs, "sql_type");
+    let any_sql_types = !sql_types.is_empty();
```

Comment:
What's the point of `AsExpression` without any sql_type ? Should we maybe make this an error instead ?

Reply:
The point would be this:

```
#[derive(Debug, AsExpression)]
#[cfg_attr(something, sql_type = "SomeType")]
struct Foo;
```

If we error here, we force them to write

```
#[derive(Debug)]
#[cfg_attr(something, derive(AsExpression))]
#[cfg_attr(something, sql_type = "SomeType")]
struct Foo;
```

An example of this in Diesel is:

```
    #[derive(FromSqlRow, AsExpression)]
    #[diesel(foreign_derive)]
    #[cfg_attr(feature = "mysql", sql_type = "::sql_types::Tinyint")]
    struct I8Proxy(i8);
```

Reply:
Also it takes more code to error here than to generate no code. ;)